Clients/Xamarin.Interactive.Client/Telemetry/Client.cs

```diff
@@ -101,141 +76,50 @@ void ObservePreferenceChange (PreferenceChange change)
                 enabled = Prefs.Telemetry.Enabled.GetValue ();
                 if (!enabled) {
                     Log.Info (TAG, "Telemetry disabled.");
-                    httpClient = null;
-                } else if (httpClient == null) {
+                    appInsightsClient = null;
+                } else if (appInsightsClient == null) {
                     Log.Info (TAG, "Telemetry will be enabled after restart.");
                 }
             }
         }
 
-        public void Post (ITelemetryEvent evnt)
+        public void Post (Exception exception)
         {
-            if (httpClient == null || !enabled)
+            if (appInsightsClient == null || !enabled)
                 return;
 
-            PostEventAsync (evnt).Forget ();
-        }
-
-        async Task PostEventAsync (ITelemetryEvent evnt)
-        {
-#pragma warning disable 0168
-            for (int i = 0; i < 10; i++) {
-                try {
-                    await PostEventOnceAsync (evnt);
-                    return;
-                } catch (JsonSerializationException e) {
-                    Log.Error (TAG, e);
-                    return;
-                } catch (Exception e) {
-#if DEBUG
-                    Log.Error (TAG, $"attempt {i} failed", e);
-#endif
-                }
-
-                await Task.Delay (TimeSpan.FromMilliseconds (Math.Pow (2, i) * 100));
-#pragma warning restore 0168
+            try {
+                appInsightsClient.TrackException (
+                    exception,
+                    commonProperties);
+                appInsightsClient.Flush ();
+            } catch (Exception e) {
+                Log.Error (TAG, e);
             }
         }
 
-        async Task PostEventOnceAsync (ITelemetryEvent evnt)
+        public void Post (ITelemetryEvent evnt)
         {
-            var response = await httpClient.PostAsync (
-                "logEvent",
-                new EventObjectStreamContent (this, evnt));
+            if (appInsightsClient == null || !enabled)
+                return;
 
-            if (response.StatusCode == HttpStatusCode.Forbidden) {
-                Log.Info (TAG, "Disabling telemetry at service's direction (403).");
-                DisableTelemetry ();
+            try {
+                var props = new Dictionary<string, string> (commonProperties);
+                foreach (var pair in evnt.GetProperties ())
+                    props [pair.Key] = pair.Value;
+
+                appInsightsClient.TrackEvent (
+                    ProducerNamespace + evnt.GetType ().Name,
+                    properties: props);
+                appInsightsClient.Flush ();
+            } catch (Exception e) {
+                Log.Error (TAG, e);
```

Comment:
You do _not_ want to log exceptions like this from telemetry. This is likely to result in an infinite exception loop of death. There is actually a flag on `LogEntry` that should prevent exceptions from being sent as telemetry, so set that flag?

Reply:
Cool, I missed that flag.